File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, bool> map;
        if(nums.size() == 0) {return 0;}
        
        int max_fin = 0;
        for(int i = 0; i < nums.size(); i++) {
            map[nums[i]] = true;
        }
        for(int i = 0; i < nums.size(); i++) {
                        int temp_max = 1;

            if(map[nums[i]-1] == true) {
                
            }
            else{
            
            while(map[nums[i]+temp_max] == true) {
                temp_max++;
            }
            }
            max_fin = max(temp_max, max_fin);
        }
        return max_fin;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        my_set = set()
        max_seq = 0
        
        for num in nums:
            my_set.add(num)
            
        for num in nums:
            
            
            if num-1 in my_set:
                continue
            else:
                temp = num
                count = 1
                while temp+1 in my_set:
                    count += 1
                    temp += 1
                max_seq = max(count, max_seq)
                
        return max_seq
            
                
            
            
            
            
        
            
        
            
        
            